pennylane/tensorbox/tf_box.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module contains the TensorFlowBox implementation of the TensorBox API.
+"""
+import tensorflow as tf
+
+import pennylane as qml
+
+
+class TensorFlowBox(qml.TensorBox):
+    """Implements the :class:`~.TensorBox` API for TensorFlow tensors.
+
+    For more details, please refer to the :class:`~.TensorBox` documentation.
+    """
+
+    @property
+    def interface(self):
+        return "tf"
+
+    def __len__(self):

```

Comment:
How come this class had to overwrite that dunder method?

Reply:
I don't think `len` is defined for tensorflow objects...was struggling with that before. 

As a good rule of thumb: wrapping has lots of nasty surprised :)

Reply:
It's worse than that --- it is defined for `tf.Tensor` but not `tf.Variable` or `tf.ResourceVariable`! So it works sometimes, but not all the time, which is really annoying for wrapping.